File: bone-platform/bone-iam/src/main/java/com/bone/iam/domain/role/Role.java
package com.bone.iam.domain.role;

import com.bone.core.domain.TenantAggregateRoot;
import com.bone.iam.domain.role.event.RoleCreatedEvent;
import com.bone.metadata.sdk.domain.annotation.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Table("iam_role")
public class Role extends TenantAggregateRoot<Long> {
    private String name;
    private String code;
    private int type;
    private String description;
    private Long parentRoleId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public static Role create(String name, String code, String description, int type, Long tenantId, Long parentRoleId) {
        Role role = new Role();
        role.name = name;
        role.code = code;
        role.description = description;
        role.type = type;
        role.setTenantId(tenantId);
        role.parentRoleId = parentRoleId;
        role.createdAt = LocalDateTime.now();
        role.updatedAt = LocalDateTime.now();
        role.addDomainEvent(new RoleCreatedEvent(role));
        return role;
    }

    public void update(String description) {
        this.description = description;
        this.updatedAt = LocalDateTime.now();
    }
}

File: bone-frontend/packages/shared-types/src/role.ts
/**
 * 角色领域类型（对齐 iam-app 真实模型）
 */
import type { Permission } from './permission';

export interface Role {
  id: number;
  tenantId: number;
  name: string;
  code: string;
  type: 0 | 1; // 0=系统角色, 1=自定义角色
  description: string;
  parentRoleId?: number;
  createdBy?: number;
  updatedBy?: number;
  createdAt: string;
  updatedAt: string;
  deleted: boolean;
  version?: number;
  permissions?: Permission[];
}

export interface CreateRoleRequest {
  name: string;
  description: string;
  code?: string;
  tenantId?: number;
  parentRoleId?: number;
}

export interface UpdateRoleRequest {
  name: string;
  description: string;
  code?: string;
  tenantId?: number;
  parentRoleId?: number;
}
